plutus-list/test/Spec.hs

```diff
@@ -0,0 +1,153 @@
+{-# LANGUAGE ViewPatterns #-}
+
+import Data.Function (on)
+import Data.Kind (Type)
+import Data.List (nub, nubBy, sort, sortOn)
+import PlutusTx.List.Natural qualified as Nat
+import PlutusTx.List.Ord (
+  isSorted,
+  isSortedMonotonic,
+  isSortedMonotonicOn,
+  isSortedOn,
+ )
+import PlutusTx.Natural (Natural)
+import PlutusTx.Prelude qualified as PTx
+import Test.Tasty (TestTree, defaultMain, localOption, testGroup)
+import Test.Tasty.QuickCheck (
+  Arbitrary (arbitrary, shrink),
+  Fun (Fun),
+  Gen,
+  Property,
+  QuickCheckTests,
+  Testable (property),
+  checkCoverage,
+  cover,
+  forAllShrink,
+  oneof,
+  testProperty,
+ )
+
+main :: IO ()
+main = defaultMain $ testGroup "Tests" tests
+
+tests :: [TestTree]
+tests =
+  [ localOption go
+      . testGroup "List functions with Natural"
+      $ [ testProperty "length == PlutusTx.length" propLengthAgree
+        , testProperty "take == PlutusTx.take" propTakeAgree
+        , testProperty "take n xs ++ drop n xs == xs" propTakeDrop
+        , testProperty "splitAt n xs == (take n xs, drop n xs)" propSplitAt
+        , testProperty "length (replicate n x) == n" propReplicateLength
+        , testProperty "forall y in (replicate n x), x == y" propReplicateElem
+        ]
+  , testGroup
+      "List sortedness"
+      [ testProperty "isSorted xs == (sort xs == xs)" propIsSorted
+      , testProperty "isSortedMonotonic xs == (nub (sort xs) == xs)" propIsSortedMonotonic
+      , testProperty "isSortedOn f xs == isSorted (f <$> xs)" propIsSortedOn
+      , testProperty "isSortedMonotonicOn f xs == isSortedMonotonic (f <$> xs)" propIsSortedMonotonicOn
+      ]
+  ]
+  where
+    go :: QuickCheckTests
+    go = 100000
+
+-- This ensures at least 50% sorted lists.
+genSomeSortedListOn :: (Arbitrary a, Ord b) => (a -> b) -> Gen (Either [a] [a])
```

Comment:
I would actually wrap these generators and shrinkers by using newtypes, with `Arbitrary` instances over them.